### PR TITLE
Saving Kotlin classfiles properly

### LINKED ISSUE #391: Kotlin compilation fails if TestSpark installed locally: `permission denied` when `kotlinc` writes `.class`-files
**Describe the bug**
TODO: add description.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by Hello-zoka**
# Description of changes made
`kotlinc` saves `.class` files in the same place they were called from, but we expect them to be in the same directory as source files. 

Probably, coverage collection for Kotlin in headless mode works now(has to be tested)

Closes #391

### CODE REVIEW

File: core/src/main/kotlin/org/jetbrains/research/testspark/core/test/kotlin/KotlinTestCompiler.kt [hunk 1]
```diff
@@ -64,13 +64,20 @@ class KotlinTestCompiler(
...
+                "-d",
+                workingDir
             ),
         )
 
         logger.info { "Error message: '$errorMsg'" }
 
-        // No need to save the .class file for kotlin, so checking the error message is enough
-        return Pair(errorMsg.isBlank(), errorMsg)
+        val classFilePath = path.replace(".kt", ".class")
```
Comment (Vladislav0Art) [inline core/src/main/kotlin/org/jetbrains/research/testspark/core/test/kotlin/KotlinTestCompiler.kt:77]: Are "dirty" file names like `MyKotlinClass.kt.kt` possible? 
If so, then "replace last" semantic is safer here.

Comment (Hello-zoka) [reply core/src/main/kotlin/org/jetbrains/research/testspark/core/test/kotlin/KotlinTestCompiler.kt:77]: Unfortunately, there is no removeLast in Kotlin. Fixed.

**Review by pderakhshanfar — APPROVED** [review]
I tested it on IDEA 2024.2.3, and the problem seems to be resolved.

**Comment by Vladislav0Art** [comment]
I got a compilation error but I think it is a **local issue with my IntelliJ**:
```
2024-10-15 11:23:09,907 [  82397]   INFO - STDOUT - 11:23:09.907 [ApplicationImpl pooled thread 6] INFO org.jetbrains.research.testspark.core.test.kotlin.KotlinTestCompiler - Error message: 'exception: java.lang.OutOfMemoryError: Java heap space	at java.base/java.util.zip.ZipFile$Source.initCEN(ZipFile.java:1678)	at java.base/java.util.zip.ZipFile$Source.<init>(ZipFile.java:1495)	at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1458)	at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:724)	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:251)	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:180)	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:194)	at com.intellij.openapi.vfs.impl.ZipHandler$1.createAccessor(ZipHandler.java:22)	at com.intellij.openapi.vfs.impl.ZipHandler$1.createAccessor(ZipHandler.java:15)	at com.intellij.util.io.FileAccessorCache.createHandle(FileAccessorCache.java:45)	at com.intellij.util.io.FileAccessorCache.get(FileAccessorCache.java:39)	at com.intellij.openapi.vfs.impl.ZipHandler.acquireZipHandle(ZipHandler.java:46)	at com.intellij.openapi.vfs.impl.ZipHandlerBase.contentsToByteArray(ZipHandlerBase.java:100)	at com.intellij.openapi.vfs.impl.jar.CoreJarVirtualFile.contentsToByteArray(CoreJarVirtualFile.java:111)	at org.jetbrains.kotlin.cli.jvm.compiler.JvmPackagePartProvider$addRoots$1.invoke(JvmPackagePartProvider.kt:56)	at org.jetbrains.kotlin.cli.jvm.compiler.JvmPackagePartProvider$addRoots$1.invoke(JvmPackagePartProvider.kt:55)	at org.jetbrains.kotlin.cli.jvm.compiler.JvmPackagePartProviderKt.tryLoadModuleMapping(JvmPackagePartProvider.kt:73)	at org.jetbrains.kotlin.cli.jvm.compiler.JvmPackagePartProvider.addRoots(JvmPackagePartProvider.kt:55)	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.createPackagePartProvider(KotlinCoreEnvironment.kt:303)	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1$1.invoke(KotlinToJVMBytecodeCompiler.kt:261)	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1$1.invoke(KotlinToJVMBytecodeCompiler.kt:261)	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.createContainer(TopDownAnalyzerFacadeForJVM.kt:187)	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.createContainer$default(TopDownAnalyzerFacadeForJVM.kt:126)	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:90)	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:77)	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:256)	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:247)	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:115)	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:247)	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:87)	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:43)	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:165)'
```
Having `java.lang.OutOfMemoryError: Java heap space` is weird.

**Review by Vladislav0Art — APPROVED** [review]
👍

**Comment by Vladislav0Art** [comment]
**Requires publishing a new version of the core.** 

I see several PRs need a new major version of the core. How about merging the PRs and creating a separate PR specifically to publish a major core version at the end (for the sake of having a smaller number)?

CC: @pderakhshanfar @arksap2002 @Hello-zoka

**Pull request merged**